src/containers/About/About.js

```diff
@@ -0,0 +1,182 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { Table } from '@tektoncd/dashboard-components';
+import { InlineNotification } from 'carbon-components-react';
+import { getInstallProperties } from '../../api';
+import { isWebSocketConnected } from '../../reducers';
+
+const initialState = {
+  dashboardInfo: [],
+  isLoaded: false,
+  isNotFinished: true,
+  error: ''
+};
+
+export /* istanbul ignore next */ class About extends Component {
+  constructor(props) {
+    super(props);
+    this.state = initialState;
+  }
+
+  componentDidMount() {
+    this.fetchDashboardInfo();
+  }
+
+  setErrorState(errorsFound) {
+    let errorMessage = 'Error getting version of ';
+
+    for (let i = 0; i < errorsFound.length; i += 1) {
+      if (i === 0) {
+        errorMessage += errorsFound[i];
+      } else if (i === errorsFound.length - 1) {
+        errorMessage += ` and ${errorsFound[i]}`;
+      } else {
+        errorMessage += `, ${errorsFound[i]}`;
+      }
+    }
+
+    if (this.state.isNotFinished) {
+      this.setState({
+        error: errorMessage,
+        isNotFinished: false
+      });
+    }
+  }
+
+  async fetchDashboardInfo() {
+    const dash = await getInstallProperties();
+    this.setState({
+      dashboardInfo: dash,
+      isLoaded: true
+    });
+  }
+
+  render() {
+    const { intl, loading } = this.props;
+    if (this.state.dashboardInfo !== null && this.state.isLoaded === true) {
+      const initialHeaders = [
+        {
+          key: 'property',
+          header: intl.formatMessage({
+            id: 'dashboard.tableHeader.Property',
+            defaultMessage: 'Property'
+          })
+        },
+        {
+          key: 'value',
+          header: intl.formatMessage({
+            id: 'dashboard.tableHeader.Value',
+            defaultMessage: 'Value'
+          })
+        }
+      ];
+
+      const initialRows = [];
+      let data = '';
+      const errorsToDisplay = [];
+
+      if (
+        this.state.dashboardInfo.InstallNamespace !== '' &&
+        this.state.dashboardInfo.InstallNamespace !== 'undefined'
+      ) {
+        data = {
+          id: `InstallNamespace`,
+          property: 'Install Namespace',
+          value: this.state.dashboardInfo.InstallNamespace
+        };
+        initialRows.push(data);
+      } else {
+        errorsToDisplay.push('Install Namespace');
+      }
+
+      if (
```

Comment:
Can we reduce the duplication here? We have essentially the same block repeated 3 times, once for each of the properties.